firecracker/rootfs/scripts/create_rootfs_image.sh

```diff
@@ -21,7 +21,9 @@ readonly IMAGE="${BUILD_DIR}/${IMAGE_NAME}.ext4"
 readonly MOUNT_POINT="${BUILD_DIR}/mount_point"
 mkdir -p "${MOUNT_POINT}"
 
+# Declare that we need these env vars later
 readonly SIZE_MB="${SIZE_MB}"
+readonly PLUGIN_BOOTSTRAP_INIT_ARTIFACTS_DIR="${PLUGIN_BOOTSTRAP_INIT_ARTIFACTS_DIR}"
```

Comment:
Basically I was trying to declare at the very top "here are all the variables you need to send in" instead of having hat happen on line 50 or whatever

Reply:
but yeah READONLY seemingly reads it and fails if its not there, so, great

Reply:
Just for completeness, you can also do this as a way to front-load your "absolutely required inputs"

```bash
: ${SIZE_MB}
```
The `:` is the "no-op" command, and it's just there (in this case) to trigger a failure if `${SIZE_MB}` is unset. You'll see this pattern pop up not infrequently.

It is pretty damn un-Googleable, though 🤷 

https://tldp.org/LDP/abs/html/special-chars.html